src/main/java/org/joshsim/lang/interpret/visitor/JoshConfigDiscoveryVisitor.java

```diff
@@ -0,0 +1,136 @@
+    if (childResult != null) {
+      result.addAll(childResult);
+    }
+
+    return result;
+  }
+
+  @Override
+  public Set<DiscoveredConfigVar> visitConfigValueWithDefault(
+      JoshLangParser.ConfigValueWithDefaultContext ctx) {
+    String variableName = ctx.name.getText();
+    String defaultValueString = extractDefaultValueString(ctx.defaultValue);
+    DiscoveredConfigVar discovered = new DiscoveredConfigVar(variableName, defaultValueString);
+    Set<DiscoveredConfigVar> result = new HashSet<>();
+    result.add(discovered);
+
+    // Continue visiting children to collect any nested config values
+    Set<DiscoveredConfigVar> childResult = visitChildren(ctx);
+    if (childResult != null) {
+      result.addAll(childResult);
+    }
+
+    return result;
+  }
+
+  /**
+   * Extracts a readable string representation from a default value expression.
+   *
+   * <p>This method traverses the parse tree to build a human-readable string
+   * representation of the default value, handling common cases like numbers
+   * with units, strings, and simple expressions.</p>
+   *
+   * @param defaultValueContext The parse tree context for the default value expression
+   * @return A string representation of the default value
+   */
+  private String extractDefaultValueString(ParseTree defaultValueContext) {
+    if (defaultValueContext == null) {
+      return "";
+    }
+
+    // Handle simple cases based on context type using new-style switch
+    switch (defaultValueContext.getClass().getSimpleName()) {
+      case "SimpleExpressionContext":
+        JoshLangParser.SimpleExpressionContext simpleCtx =
+            (JoshLangParser.SimpleExpressionContext) defaultValueContext;
+        return extractUnitsValueString(simpleCtx.unitsValue());
+      case "SimpleNumberContext":
+        JoshLangParser.SimpleNumberContext numberCtx =
+            (JoshLangParser.SimpleNumberContext) defaultValueContext;
+        return numberCtx.number().getText();
+      case "SimpleStringContext":
+        JoshLangParser.SimpleStringContext stringCtx =
+            (JoshLangParser.SimpleStringContext) defaultValueContext;
+        return stringCtx.string().STR_().getText();
+      default:
+        // For complex expressions, fall back to getting the full text
+        return defaultValueContext.getText();
+    }
+  }
+
+  /**
+   * Extracts a string representation from a units value (number + optional units).
+   *
+   * @param unitsValueContext The parse tree context for the units value
+   * @return A string representation like "5 m" or "10"
+   */
+  private String extractUnitsValueString(JoshLangParser.UnitsValueContext unitsValueContext) {
+    if (unitsValueContext == null) {
+      return "";
+    }
+
+    StringBuilder result = new StringBuilder();
+    result.append(unitsValueContext.number().getText());
+
+    // Check for units - could be identifier or PERCENT_
+    if (unitsValueContext.identifier() != null) {
```

Comment:
Is there a way that unitsValueContext can accept this instead of doing this kind of inspection?